tests/GovUk.Frontend.AspNetCore.ConformanceTests/ComponentTests.cs

```diff
@@ -39,16 +55,110 @@ private static void AppendToDescribedBy(ref string describedBy, string value)
         }
     }
 
+
     private void CheckComponentHtmlMatchesExpectedHtml<TOptions>(
         ComponentTestCaseData<TOptions> testCaseData,
         Func<ComponentGenerator, TOptions, string> generateComponent,
         Predicate<IDiff> excludeDiff = null)
     {
         var html = generateComponent(_componentGenerator, testCaseData.Options);
 
+        // Retrieve the currently executing test method and class dynamically
+        var testClass = GetCurrentTestClassName();
+        var testMethod = GetCurrentTestMethodName();
+        var testMethodData = testCaseData.Name;
+        var testMethodDisplay = $"{testClass}__{testMethod}--{testMethodData}";
+        var testMethodDisplaySafe = string.Join("_", testMethodDisplay.Split(Path.GetInvalidFileNameChars()));
+
+        // Base directory for output
+        var outputBaseDirectory = Path.Combine(Path.GetTempPath(), "govuk-frontend-aspnetcore");
+        var expectedFilePath = Path.Combine(outputBaseDirectory, $"{testMethodDisplaySafe}--expected--{_govUkFrontendVersion}.html");
+        var actualFilePath = Path.Combine(outputBaseDirectory, $"{testMethodDisplaySafe}--actual--{_govUkFrontendVersion}.html");
+
+        try
+        {
+            // Create base directory if it doesn't exist
+            Directory.CreateDirectory(outputBaseDirectory);
+
+            // Write the expected and actual HTML to files for easier diffing later
+            File.WriteAllText(expectedFilePath, testCaseData.ExpectedHtml);
+            File.WriteAllText(actualFilePath, html);
+
+            _outputHelper.WriteLine($"Expected HTML written to: {expectedFilePath}");
+            _outputHelper.WriteLine($"Actual HTML written to: {actualFilePath}");
+        }
+        catch (Exception ex)
+        {
+            throw new InvalidOperationException($"Failed to write HTML files to {outputBaseDirectory}.", ex);
+        }
+
+        // Assert that the generated HTML matches the expected HTML
         AssertEx.HtmlEqual(testCaseData.ExpectedHtml, html, excludeDiff);
     }
```

Comment:
This is some exploratory code where I was writing files to disk (found it easier to diff, and to try and pinpoint what changed/when):

![image](https://github.com/user-attachments/assets/9796923d-8ca0-4ba5-af47-bdbc1182738b)


There is potentially some utility in storing these within the repo, rather than a temp directory, perhaps? (or maybe just the `Expected` files as this is what is supplied from upstream for each version?)

That's probably a separate discussion/PR though, and I suspect we'll want to avoid merging this for now?